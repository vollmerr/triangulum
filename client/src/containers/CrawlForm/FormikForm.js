import React from 'react';
import {
  Dimmer,
  Loader,
  Button,
  Form,
} from 'semantic-ui-react';

import schema from './schema';
import withFormik from './withFormik';

import Field from '../../components/Form/Field';
import ErrorMessage from '../../components/Form/ErrorMessage';

/**
 * Form to wrap with formik HOC
 */
class FormikForm extends React.PureComponent {
  renderFields = () => (
    Object.keys(schema).map(field => (
      <Field key={field} formikProps={this.props} field={schema[field]} />
    ))
  )

  render() {
    const {
      handleSubmit,
      isSubmitting,
    } = this.props;

    const submitProps = {
      fluid: true,
      icon: 'send',
      size: 'large',
      type: 'submit',
      color: 'violet',
      content: 'Send',
      disabled: isSubmitting,
    };

    return (
      <div>
        <Dimmer active={isSubmitting} inverted>
          <Loader>Crawling the interwebs...</Loader>
        </Dimmer>

        <Form size={'large'} onSubmit={handleSubmit}>
          {this.renderFields()}
          <Button {...submitProps} />
        </Form>

        <ErrorMessage schema={schema} {...this.props} />
      </div>
    );
  }
}

export default withFormik(FormikForm);
