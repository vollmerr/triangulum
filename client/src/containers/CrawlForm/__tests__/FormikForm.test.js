import React from 'react';
import { shallow } from 'enzyme';

import { FormikForm } from '../FormikForm';

const props = {
  handleSubmit: jest.fn(),
  isSubmitting: false,
};

let wrapper;
describe('FormikForm', () => {
  beforeEach(() => {
    wrapper = shallow(<FormikForm {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
