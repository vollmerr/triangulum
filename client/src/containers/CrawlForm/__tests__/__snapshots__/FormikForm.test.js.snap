// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormikForm should render correctly 1`] = `
<Container
  className="page"
  text={true}
>
  <Dimmer
    active={false}
    inverted={true}
  >
    <Loader
      data-cy="loader"
    >
      Crawling the interwebs...
    </Loader>
  </Dimmer>
  <Form
    as="form"
    onSubmit={[MockFunction]}
  >
    <Field
      data-cy="field-url"
      field={
        Object {
          "control": [Function],
          "label": "Start URL",
          "name": "url",
          "placeholder": "http://example.com",
        }
      }
      formikProps={
        Object {
          "handleSubmit": [MockFunction],
          "isSubmitting": false,
        }
      }
      key="url"
    />
    <Field
      data-cy="field-type"
      field={
        Object {
          "control": [Function],
          "label": "Search Type",
          "name": "type",
          "options": Array [
            Object {
              "text": "Breadth-first",
              "value": "bfs",
            },
            Object {
              "text": "Depth-first",
              "value": "dfs",
            },
          ],
          "placeholder": "Select a Type",
        }
      }
      formikProps={
        Object {
          "handleSubmit": [MockFunction],
          "isSubmitting": false,
        }
      }
      key="type"
    />
    <Field
      data-cy="field-limit"
      field={
        Object {
          "control": [Function],
          "label": "Limit",
          "name": "limit",
          "placeholder": "Maximum Depth Limit",
          "type": "number",
        }
      }
      formikProps={
        Object {
          "handleSubmit": [MockFunction],
          "isSubmitting": false,
        }
      }
      key="limit"
    />
    <Field
      data-cy="field-keyword"
      field={
        Object {
          "control": [Function],
          "label": "Keyword (optional)",
          "name": "keyword",
          "placeholder": "Word to search for",
        }
      }
      formikProps={
        Object {
          "handleSubmit": [MockFunction],
          "isSubmitting": false,
        }
      }
      key="keyword"
    />
    <ErrorMessage
      handleSubmit={[MockFunction]}
      isSubmitting={false}
      schema={
        Object {
          "keyword": Object {
            "control": [Function],
            "label": "Keyword (optional)",
            "name": "keyword",
            "placeholder": "Word to search for",
          },
          "limit": Object {
            "control": [Function],
            "label": "Limit",
            "name": "limit",
            "placeholder": "Maximum Depth Limit",
            "type": "number",
          },
          "type": Object {
            "control": [Function],
            "label": "Search Type",
            "name": "type",
            "options": Array [
              Object {
                "text": "Breadth-first",
                "value": "bfs",
              },
              Object {
                "text": "Depth-first",
                "value": "dfs",
              },
            ],
            "placeholder": "Select a Type",
          },
          "url": Object {
            "control": [Function],
            "label": "Start URL",
            "name": "url",
            "placeholder": "http://example.com",
          },
        }
      }
    />
    <Button
      as="button"
      className="submitBtn"
      color="violet"
      content="Send"
      data-cy="button-submit"
      disabled={false}
      fluid={true}
      icon="send"
      role="button"
      type="submit"
    />
  </Form>
</Container>
`;
